Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Instance/InstanceDomGenerator.cpp

```diff
@@ -126,6 +135,77 @@ namespace AzToolsFramework
+            }
+
+            InstanceOptionalReference focusedInstanceRef = prefabFocusInterface->GetFocusedPrefabInstance(s_editorEntityContextId);
+            if (!focusedInstanceRef.has_value())
+            {
+                AZ_Assert(false, "Prefab - InstanceDomGenerator::GenerateEntityDom - "
+                    "Could not get the focused instance. It should not be null.");
+                return false;
+            }
+            const Instance& focusedInstance = focusedInstanceRef->get();
+            const Instance* focusedInstancePtr = &focusedInstance;
+
+            const AZ::EntityId entityId = entity.GetId();
+            InstanceOptionalReference owningInstanceRef = m_instanceEntityMapperInterface->FindOwningInstance(entityId);
+            if (!owningInstanceRef.has_value())
+            {
+                AZ_Assert(false, "Prefab - InstanceDomGenerator::GenerateEntityDom - "
+                    "Could not get the owning instance for the given entity id.");
+                return false;
+            }
+            const Instance& owningInstance = owningInstanceRef->get();
+
+            // Climbs up from the owning instance to root instance, but stops at the focused instance if they can meet.
+            const InstanceClimbUpResult climbUpResult = PrefabInstanceUtils::ClimbUpToTargetOrRootInstance(owningInstance, focusedInstancePtr);
+            const Instance* focusedOrRootInstancePtr = climbUpResult.m_reachedInstance;
+            if (!focusedOrRootInstancePtr)
+            {
+                AZ_Assert(false, "Prefab - InstanceDomGenerator::GenerateEntityDom - "
+                    "Could not get the focused or root instance from climb-up. It should not be null.");
+                return false;
+            }
+
+            // Prepares the path from the focused (or root) instance to the entity.
+            AZStd::string relativePathFromTop = PrefabInstanceUtils::GetRelativePathFromClimbedInstances(climbUpResult.m_climbedInstances);
+            AZStd::string entityAliasPath = m_instanceToTemplateInterface->GenerateEntityAliasPath(entityId);
+            AZStd::string relativePathFromTopToEntity = relativePathFromTop + entityAliasPath;
+            PrefabDomPath relativeDomPathFromTopToEntity(relativePathFromTopToEntity.c_str());
+
+            // Retrieves the entity DOM from the tempalte of the focused or root instance.
+            PrefabDom focusedOrRootTemplateDom;
+            focusedOrRootTemplateDom.CopyFrom(m_prefabSystemComponentInterface->FindTemplateDom(focusedOrRootInstancePtr->GetTemplateId()),
+                entityDom.GetAllocator());
+
+            // If it is the focused container entity, then replace the entity DOM's transform data (from template)
+            // with the one seen from root.
+            if (entityId == focusedInstancePtr->GetContainerEntityId())
+            {
+                UpdateContainerEntityInDomFromRoot(focusedOrRootTemplateDom, focusedInstance);
```

Comment:
Food for thought: This can be a follow up task but in both GenerateInstanceDom and GenerateEntityDom, we are repeating the same logic and copying the dom over and over to update the transform of the focused container. Instead of doing it this way, maybe we can just calculate it once when the focus changes and reuse that dom instead of doing a log of copies everytime either GenerateInstanceDom or GenerateEntityDom is called. That one-time calculated dom can live in one of the focus handlers that tracks the focused instance.

@forkercat , can you create a backlog task for this?

Reply:
this is actually a good idea. It doesn't make sense to climb up to the root to get this info every time it needs. I'll create a task.

It sounds like an implementation using an unordered map <instance, transform DOM/container entity DOM>. GET is straightforward. But I think we need to have some forethought about how we SET the transform data in map correctly when it changes.

Reply:
It doesn't have to be a map. It can just be a single dom that is stored in the focus handler. If the focus changes, the dom will be updated accordingly. The dom can be that of the focused instance or just the container entity of the focused instance depending on how we'll implement the copying of doms.